# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/kwin/hotfix-wayland.patch
# Copyright (C) 2024 - 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- kwin-6.0.4/src/wayland/outputdevice_v2.cpp.vanilla	2024-04-28 15:06:35.861000000 +0200
+++ kwin-6.0.4/src/wayland/outputdevice_v2.cpp	2024-04-28 15:07:26.558000000 +0200
@@ -446,23 +446,29 @@
 
 void OutputDeviceV2InterfacePrivate::sendBrightnessMetadata(Resource *resource)
 {
+#if 0
     if (resource->version() >= KDE_OUTPUT_DEVICE_V2_BRIGHTNESS_METADATA_SINCE_VERSION) {
         send_brightness_metadata(resource->handle, std::round(m_maxPeakBrightness.value_or(0)), std::round(m_maxAverageBrightness.value_or(0)), std::round(m_minBrightness * 10'000));
     }
+#endif
 }
 
 void OutputDeviceV2InterfacePrivate::sendBrightnessOverrides(Resource *resource)
 {
+#if 0
     if (resource->version() >= KDE_OUTPUT_DEVICE_V2_BRIGHTNESS_OVERRIDES_SINCE_VERSION) {
         send_brightness_overrides(resource->handle, std::round(m_maxPeakBrightnessOverride.value_or(-1)), std::round(m_maxAverageBrightnessOverride.value_or(-1)), std::round(m_minBrightnessOverride.value_or(-0.000'1) * 10'000));
     }
+#endif
 }
 
 void OutputDeviceV2InterfacePrivate::sendSdrGamutWideness(Resource *resource)
 {
+#if 0
     if (resource->version() >= KDE_OUTPUT_DEVICE_V2_SDR_GAMUT_WIDENESS_SINCE_VERSION) {
         send_sdr_gamut_wideness(resource->handle, std::clamp<uint32_t>(m_sdrGamutWideness * 10'000, 0, 10'000));
     }
+#endif
 }
 
 void OutputDeviceV2Interface::updateGeometry()
--- kwin-6.2.5/src/wayland/outputmanagement_v2.cpp.vanilla	2025-02-04 21:49:12.423333312 +0100
+++ kwin-6.2.5/src/wayland/outputmanagement_v2.cpp	2025-02-04 21:49:32.593333312 +0100
@@ -64,8 +64,8 @@
     void kde_output_configuration_v2_set_wide_color_gamut(Resource *resource, wl_resource *outputdevice, uint32_t enable_wcg) override;
     void kde_output_configuration_v2_set_auto_rotate_policy(Resource *resource, wl_resource *outputdevice, uint32_t auto_rotation_policy) override;
     void kde_output_configuration_v2_set_icc_profile_path(Resource *resource, wl_resource *outputdevice, const QString &profile_path) override;
-    void kde_output_configuration_v2_set_brightness_overrides(Resource *resource, wl_resource *outputdevice, int32_t max_peak_brightness, int32_t max_average_brightness, int32_t min_brightness) override;
-    void kde_output_configuration_v2_set_sdr_gamut_wideness(Resource *resource, wl_resource *outputdevice, uint32_t gamut_wideness) override;
+    void kde_output_configuration_v2_set_brightness_overrides(Resource *resource, wl_resource *outputdevice, int32_t max_peak_brightness, int32_t max_average_brightness, int32_t min_brightness);
+    void kde_output_configuration_v2_set_sdr_gamut_wideness(Resource *resource, wl_resource *outputdevice, uint32_t gamut_wideness);
     void kde_output_configuration_v2_set_color_profile_source(Resource *resource, wl_resource *outputdevice, uint32_t source) override;
     void kde_output_configuration_v2_set_brightness(Resource *resource, wl_resource *outputdevice, uint32_t brightness) override;
 };
